hil/api.py

```diff
@@ -690,18 +690,24 @@ def headnode_detach_network(headnode, hnic):
 @rest_call('GET', '/networks', Schema({}))
 def list_networks():
     """Lists all networks"""
-
-    get_auth_backend().require_admin()
-
     networks = db.session.query(model.Network).all()
     result = {}
-    for n in networks:
-        if n.access:
-            net = {'network_id': n.network_id,
-                   'projects': sorted([p.label for p in n.access])}
-        else:
-            net = {'network_id': n.network_id, 'projects': None}
-        result[n.label] = net
+    # Admin Operation
+    if get_auth_backend().have_admin():
+        for n in networks:
+            if n.access:
+                net = {'network_id': n.network_id,
+                       'projects': sorted([p.label for p in n.access])}
+            else:
+                net = {'network_id': n.network_id, 'projects': None}
+            result[n.label] = net
+    # Regular User Operation
+    else:
+        pub_nets = db.session.query(model.Network).filter_by(owner=None).all()
+        for n in pub_nets:
+            if not n.access:
```

Comment:
I think what we want to do here is actually `filter_by(access=None)`; owner is about who can delete the network, not access it, so we don't need to filter on that. And it's better to have the database check conditions like this, rather than fetching all the results and then filtering them in python.

We should also move first query in this function inside of the admin-only branch, since we don't use it in the non-admin case.

Reply:
make sense!